File: Programming EF Code First/chapter3/Chapter 3 VB/Fluent/DataAccess/BreakAwayContext.vb
Imports System.Data.Entity
Imports Model

Namespace DataAccess
  Public Class BreakAwayContext
	  Inherits DbContext
	Public Property Destinations() As DbSet(Of Destination)
	Public Property Lodgings() As DbSet(Of Lodging)
	Public Property Trips() As DbSet(Of Trip)
	Public Property People() As DbSet(Of Person)

	Protected Overrides Sub OnModelCreating(ByVal modelBuilder As DbModelBuilder)
	  ' Entity Type Configuration
	  modelBuilder.Configurations.Add(New DestinationConfiguration())
	  modelBuilder.Configurations.Add(New LodgingConfiguration())
	  modelBuilder.Configurations.Add(New TripConfiguration())
	  modelBuilder.Configurations.Add(New PersonConfiguration())

	  ' Complex Type Configuration
	  modelBuilder.Configurations.Add(New AddressConfiguration())
	  modelBuilder.ComplexType(Of PersonalInfo)()
	End Sub
  End Class
End Namespace

File: Programming EF Code First/chapter3/Chapter3/BAGA_FluentAPI/DataAccess/BreakAwayContext.cs
using System.Data.Entity;
using Model;

namespace DataAccess
{
  public class BreakAwayContext : DbContext
  {
    public DbSet<Destination> Destinations { get; set; }
    public DbSet<Lodging> Lodgings { get; set; }
    public DbSet<Trip> Trips { get; set; }
    public DbSet<Person> People { get; set; }

    protected override void OnModelCreating(DbModelBuilder modelBuilder)
    {
      // Entity Type Configuration
      modelBuilder.Configurations.Add(new DestinationConfiguration());
      modelBuilder.Configurations.Add(new LodgingConfiguration());
      modelBuilder.Configurations.Add(new TripConfiguration());
      modelBuilder.Configurations.Add(new PersonConfiguration());

      // Complex Type Configuration
      modelBuilder.Configurations.Add(new AddressConfiguration());
      modelBuilder.ComplexType<PersonalInfo>();
    }
  }
}